catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/CatalogMigratableImpl.java

```diff
@@ -76,46 +73,61 @@
      * @param framework     framework used to retrieve the metacards to export
      * @param filterBuilder builder used to create query filters
-     * @param fileWriter    object used to write metacards to disk
      * @param config        export configuration information
+     * @param version       version of this migratable
+     * @param id            id of this migratable
+     * @param title         title of this migratable
+     * @param organization  organization of this migratable
      */
-    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+    public CatalogMigratableImpl(@NotNull CatalogFramework framework,
             @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
-            @NotNull CatalogMigratableConfig config) {
-        this(description,
-                framework,
+            @NotNull CatalogMigratableConfig config, @NotNull String version, @NotNull String id,
+            @NotNull String title, @NotNull String description, @NotNull String organization) {
+
+        this(framework,
                 filterBuilder,
                 fileWriter,
                 config,
-                new MigrationTaskManager(config, fileWriter));
+                new MigrationTaskManager(config, fileWriter),
+                version,
+                id,
+                title,
+                description,
+                organization);
+
     }
 
     /**
      * Constructor that allows the caller choice of {@link MigrationTaskManager} to be injected.
      *
-     * @param description   description of this migratable
      * @param framework     framework used to retrieve the metacards to export
      * @param filterBuilder builder used to create query filters
-     * @param fileWriter    object used to write metacards to disk
      * @param config        export configuration information
      * @param taskManager   the manager responsible for submitting file writing jobs during export
+     * @param version       version of this migratable
+     * @param id            id of this migratable
+     * @param title         title of this migratable
+     * @param description   description of this migratable
+     * @param organization  organization of this migratable
      */
-    public CatalogMigratableImpl(@NotNull String description, @NotNull CatalogFramework framework,
+    public CatalogMigratableImpl(@NotNull CatalogFramework framework,
             @NotNull FilterBuilder filterBuilder, @NotNull MigrationFileWriter fileWriter,
-            @NotNull CatalogMigratableConfig config, @NotNull MigrationTaskManager taskManager) {
-        notNull(description, "description cannot be null");
+            @NotNull CatalogMigratableConfig config, @NotNull MigrationTaskManager taskManager,
+            @NotNull String version, @NotNull String id, @NotNull String title,
+            @NotNull String description, @NotNull String organization) {
```

Comment:
Good point. This `Describable` thing makes for very long constructors. I like the builder pattern, but it's kind of messy when you need to use it with blueprint.

If we create a `DescribableProperties` class, we'll have to add it to the `platform-api` bundle, which may raise some eyebrows. And it may make the usage a little strange. Why would you pass in a `DescribableProperties` object but have separate getters for the different pieces? Why not just have one getter that returns a `DescribableProperties` instead?

How about using setters instead of constructor arguments to initialize those?